### DISCUSSION TITLE
Support "QFileDialog.getOpenFileName" for exporting images via context menu

### DISCUSSION THREAD

**Discussion by Lexachoc** [pyqtgraph/pyqtgraph#2611] [Q&A]
Hello,
I found out that the export function uses the following way
`self.fileDialog.setFileMode(QtGui.QFileDialog.AnyFile)`
in the source code.

The file dialog style is different than using
`QFileDialog.getOpenFileName`

I use Win10 and the built-in file dialog is not so convenient for me.
(Matplotlib's toolbar "Save the figure" function use the latter native way.)

I wonder if this could be added to the export function of context menu?

I also found one example in
[https://github.com/pyqtgraph/pyqtgraph/blob/f24335dc71b66d3b2075c197367b3290e2ea4d11/pyqtgraph/examples/relativity/relativity.py](url)
which has an explicit export button that also uses the latter file dialog style.

When pyqtgraph is integrated with my own GUI (open file dialogs with `QFileDialog.getOpenFileName`), this leads to the problem of having two file dialogs with different styles, which may cause confusion.

**Comment by pijyoi** [comment]
You could use `DontUseNativeDialog` in your GUI, then the styles would be the same.
```python
from PySide6.QtWidgets import QApplication, QFileDialog
app = QApplication([])
dlg = QFileDialog.getOpenFileName(options=QFileDialog.Options.DontUseNativeDialog)
```
